services/worker-manager/schemas/v1/worker-full.yml

```diff
@@ -0,0 +1,62 @@
+$schema: "/schemas/common/metaschema.json#"
+title: Worker Full Definition
+description: A complete worker definition.
+type: object
+properties:
+
+  workerPoolId: {$ref: "worker-pool-full.json#/properties/workerPoolId"}
+
+  workerGroup:
+    title: Worker group
+    type: string
+    description: Worker group to which this worker belongs
+    minLength: {$const: identifier-min-length}
+    maxLength: {$const: identifier-max-length}
+    pattern: {$const: identifier-pattern}
+
+  workerId:
+    title: Worker ID
+    type: string
+    description: Worker ID
+    minLength: {$const: identifier-min-length}
+    maxLength: {$const: identifier-max-length}
+    pattern: {$const: identifier-pattern}
+
+  providerId:
+    title: Provider
+    type: string
+    # note that this is typically used as the workerGroup for workers,
+    # so its format should match that for workerGroup.
+    minLength: {$const: identifier-min-length}
+    maxLength: {$const: identifier-max-length}
+    pattern: {$const: identifier-pattern}
+    description: The provider responsible for managing this worker.
```

Comment:
This confused me at first, because I thought "wait, the workerPoolId should define the providerId".  But as you know that's wrong -- the worker might have been started by a different provider that has since been changed.

What would you think about adding a bit to the description to help readers avoid my confusion?

Reply:
oh, that's a good point. I did wonder about the data duplication but didn't ask about it for some reason. Yeah, I'll add that info to the description